lib/templates/dataset_binary.cpp.in

```diff
@@ -10,11 +10,14 @@
 namespace scipp::dataset {
 
 DataArray @NAME@(const DataArray &a, const DataArray &b) {
+  auto coords = union_(a.coords(), b.coords(), "@OPNAME@");
+  auto masks = union_or(a.masks(), b.masks());
+  auto attrs = intersection(a.attrs(), b.attrs());
   return DataArray(
```

Comment:
Is this moved here to ensure a specific order of execution?

Reply:
Yes. The old one did check coords before doing the computation on my machine. But to my understanding, this is compiler dependent.

Reply:
Yes, I believe the order is unspecified by the standard.